benchmark/primitives/primitives_test.go

```diff
@@ -188,6 +199,83 @@ func BenchmarkMutexWithoutDefer(b *testing.B) {
 	}
 }
 
+func BenchmarkAtomicAddInt64(b *testing.B) {
+	var c int64
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		atomic.AddInt64(&c, 1)
+	}
+	b.StopTimer()
+	if c != int64(b.N) {
+		b.Fatal("error")
+	}
+}
+
+func BenchmarkAtomicTimeValueStore(b *testing.B) {
```

Comment:
Now I'm curious how `atomic.Value` scales.  Can you use `struct { a, b int64 }` (16B) and `struct { a, b, c, d int64 }` (32B) in 2 benchmarks instead of `time.Time`?  Unless we're specifically trying to decide whether to store `Time`s as an atomic or behind a lock, it's more useful to know exactly what we're testing.

I believe what this means is that for the best performance, you should always use a pointer type for your `atomic.Value`s, and make sure the memory behind them is immutable.

Reply:
done.
Yeah, it's better to store the pointer to the variable.

Reply:
Also, knowing more now, it seems like we should just fall back to the more primitive `atomic.*Pointer` operations _always_ for the best performance.  `atomic.Value` adds a bunch of runtime type checks that don't seem useful if you know what you're doing.

It seems we don't have any benchmarks for that, though -- could you add some like these?

Maybe `AtomicPointerStore`, `AtomicTimePointerStore`, and another `b.Run` in `ValueStoreWithContention`?

Reply:
Some notes here: StorePointer vs. Store a pointer has the same per operation execution time. But StorePointer has much better (almost 2x) performance in case of contention.

Reply:
This makes sense: `Store` does a `LoadPointer`, a type check, and then a `StorePointer`.  So a ~2x difference is exactly what I was expecting to see.